Amazon.IonDotnet/Timestamp.cs

```diff
@@ -146,19 +146,47 @@ public DateTimeOffset AsDateTimeOffset()
         public Timestamp(DateTime dateTimeValue)
         {
             TimestampPrecision = Precision.Second;
-
+            FractionalSecond = 0;
             DateTimeValue = DateTime.SpecifyKind(dateTimeValue, DateTimeKind.Unspecified);
             //we have no idea about the local offset except when it's 0, so no change here
             LocalOffset = 0;
         }
 
+        /// <summary>
+        /// Initialize the timestamp with different components, offset and fractional second. Timestamps in the
+        /// binary encoding are always in UTC, while in the text encoding are in the local time. This means transcoding
+        /// requires a conversion between UTC and local time, ergo we add the offset and fractional seconds to this value.
+        /// </summary>
+        /// <param name="year">Year</param>
+        /// <param name="month">Month</param>
+        /// <param name="day">Day</param>
+        /// <param name="hour">Hour</param>
+        /// <param name="minute">Minute</param>
+        /// <param name="second">Second</param>
+        /// <param name="offset">Offset value</param>
+        /// <param name="frac">Fractional second value</param>
+        /// <param name="precision">The precision of the value</param>
+        public Timestamp(int year, int month, int day, int hour, int minute, int second, int offset, in decimal frac,
```

Comment:
Can this be an internal constructor, not part of the public API?

Reply:
Done